ckan/lib/api_token.py

```diff
@@ -124,7 +124,7 @@ def get_user_from_token(token: str,
     # do preprocessing in reverse order, allowing onion-like
     # "unwrapping" of the data, added during postprocessing, when
     # token was created
-    for plugin in _get_plugins():
+    for plugin in reversed(list(_get_plugins())):
```

Comment:
Why is this one back to reversing the old way?

Reply:
There are two sides to processing API tokens. When it's created and encoded into a string, all plugins can add additional properties to the token or modify existing properties. When the application receives the token, it decodes the token from a string and extracts all the properties. It sounds like a middleware for an API token, that's why data extraction should happen in reverse order compared to data injection. 

This line used `reversed` originally, but I misinterpret it and set reverse_order to the whole interface. After additional checks(in order to add a comment with the reason behind the reverse order for the interface), I realized, that this interface has 4 methods and 3 of them are executed in normal order and only data extraction from the token received from user happens in reverse order. That's why I restored APIToken to its original form.

Reply:
Great, can you add the reason that the reversed plugin order can't be used here to the comment above?

Reply:
Sure, done